src/api/recordlinker/routes/demo_router.py

```diff
@@ -73,3 +74,69 @@ def get_match_review_records(
     if match_review_record is None:
         raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND)
     return match_review_record
+
+
+@router.post(
+    "/record/{patient_reference_id}/link",
+    summary="Link demo records for match review",
+)
+def link_match(
+    patient_reference_id: int,
+    response: fastapi.Response,
+) -> schemas.demo.MatchReviewRecord:
+    """
+    Link demo records for match review.
+    """
+
+    match_review_record = next(
+        (d for d in data if d["incoming_record"]["patient_id"] == patient_reference_id), None
+    )
+    if match_review_record is None:
+        raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND)
+
+    # Update the linked status
+    match_review_record["linked"] = True
+    # Update the incoming record with the person_id from the potential match
+    match_review_record["incoming_record"]["person_id"] = match_review_record["potential_match"][0][
+        "person_id"
+    ]
+
+    # Save session (modifies the real response)
+    session_store.save_session(
+        response,
+        key="linked_status",
+        data=match_review_record["linked"],
```

Comment:
I believe this is going to overwrite any previous cookie value.  For example, if the user marks demo record 1 as linked, then marks demo record 2 as unlinked, the second operation will overwrite the previous data stored on record 1.

Reply:
Ah you're right. I previously had the key set to `linked_status_{patient_reference_id}` to avoid this but it was flagged as a potential issue because users could inject harmful code. However, the endpoint is typed to only handle integers as input so maybe that prevents inject issues. What do you think?

Reply:
Another way I could do it, is to keep the key as is, but change the data to be a dictionary with patient_reference_id as the key and True/False as the value like:

```
key="linked_status",
data={"1":True, "3":False}
```

Reply:
I'm liking the 2nd option better because all data is stored in a single cookie, which will make it easier to clear when we implement the reset button. And it's easier to retrieve all the data from a single cookie to render the data on the queue page.

Reply:
Agree. It makes sense for the session to be created when the user hits the `/GET /api/demo/record` endpoint for the first time. At this stage a copy of the json is created for the specific session/user and any subsequent calls to the demo endpoints retrieve, modify and update the record object in the session.

TL;DR; Line 103 should obtain the record from the session and modify it then save the updated object back. The session will contain a record at this point because it gets created the first time the user calls the `/GET /api/demo/record` endpoint.

Reply:
I modified it so that link/unlink update the linked status in the cookie (not the underlying data) and the GET `/api/demo/record` now checks the cookie for updates before returning the data to reflect any changes that were made by someone clicking link or unlink. Does that make sense?